src/sources/sequelize/hooks/get-products.hook.ts

```diff
@@ -0,0 +1,39 @@
+import {ActionRequest, ActionResponse, After, flat,} from 'adminjs';
+import {isGETMethod} from "../../../admin/admin.utils";
+import {CartModel, OrderModel, ProductModel} from "../models";
+
+export const getProducts = (): After<ActionResponse> => async (
+  response: ActionResponse,
+  request: ActionRequest,
+): Promise<ActionResponse> => {
+  if (!isGETMethod(request)) {
+    return response;
+  }
+
+  const id = parseInt(request.params.recordId, 10);
+
+  const order = await OrderModel.findByPk(id, {
+    include: [{
+      model: CartModel,
+      as: 'carts',
+      include: [{
+        model: ProductModel,
+        as: 'product'
+      }]
+    }],
+  })
+
+  const params = flat.unflatten(response.record.params);
+  params.products = order.carts?.map(cart => ({
+    quantity: cart.quantity,
+    product: {
+      id: cart.product?.id,
+      name: cart.product?.name,
+      price: cart.product?.price
+    }
+  }))
+
+  response.record.params = flat.flatten(params);
```

Comment:
Please specify types for `flatten` and `unfaltten` (since new adminjs returned value has type unknown). If you use InlteliJ you can import this patch (unzip and apply in IDE)
[flat.zip](https://github.com/SoftwareBrothers/adminjs-example-app/files/8109351/flat.zip)